### PR TITLE
feat: add extra field

### PR DESCRIPTION
**Pull request by nichmor**
Needed for  https://github.com/conda-forge/conda-smithy/issues/2034

### CODE REVIEW

File: crates/rattler_conda_types/src/package/about.rs [hunk 1]
```diff
@@ -41,6 +41,10 @@ pub struct AboutJson {
     )]
     pub doc_url: Vec<Url>,
 
+    /// Extra metadata that was passed during the build
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub extra: Option<HashMap<String, String>>,
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/package/about.rs:46]:
```suggestion
    #[serde(skip_serializing_if = "HashMap::is_empty", default)]
    pub extra: HashMap<String, String>,
```

**Review by baszalmstra — CHANGES_REQUESTED** [review]

**Comment by nichmor** [comment]
it looks like we need to use serde_json::Value as extra can contain different objects types

**Comment by baszalmstra** [comment]
I think we should use an `IndexMap` instead to preserve the insertion order.

**Comment by wolfv** [comment]
Or btree?

**Comment by nichmor** [comment]
I reused FxHashMap as we do in repodata , it is ok?

**Comment by baszalmstra** [comment]
I would prefer to just keep the insertion order using an `IndexMap`. But if you wanna sort them by name I would indeed use a `BTreeMap`.

**Comment by nichmor** [comment]
> I would prefer to just keep the insertion order using an `IndexMap`. But if you wanna sort them by name I would indeed use a `BTreeMap`.

changed to BTreeMap to not have reproducibility issues in future

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**